addons/hr_expense/tests/test_expenses_tax.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+
+from odoo import Command
+from odoo.addons.hr_expense.tests.common import TestExpenseCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestExpensesTax(TestExpenseCommon):
+
+    def test_expense_by_company_with_caba_tax(self):
+        """When using cash basis tax in an expense paid by the company, the money should transfer to the final account."""
+
+        caba_account_type = self.env['account.account.type'].create({
+            'name': 'Cash Basis',
+            'type': 'other',
+            'internal_group': 'asset',
+        })
+
+        caba_transition_account = self.env['account.account'].create({
+            'name': 'Tax CaBa Transition',
+            'user_type_id': caba_account_type.id,
+            'code': '131001',
+        })
+
+        caba_tax = self.env['account.tax'].create({
+            'name': 'tax test cash basis',
+            'tax_exigibility': 'on_payment',
+            'amount': 15,
+            'cash_basis_transition_account_id': caba_transition_account.id,
+        })
+        norm_tax = self.env['account.tax'].create({
+            'name': 'tax test normal',
+            'amount': 15,
+        })
+
+        
+        expense_sheet = self.env['hr.expense.sheet'].create({
+            'name': 'exp_comp_caba_report',
+            'employee_id': self.expense_employee.id,
+            'payment_mode': 'company_account',
+            'state': 'approve',
+            'expense_line_ids': [Command.create({
+                'name': 'exp_comp_caba',
+                'product_id': self.product_b.id,
+                'payment_mode': 'company_account',
+                'unit_amount': 20.0,
+                'employee_id': self.expense_employee.id,
+                'tax_ids': [caba_tax.id, norm_tax.id]
+            })]
+        })
+        moves = expense_sheet.action_sheet_move_create()
+        self.assertEqual(moves[expense_sheet.id].line_ids[1].account_id, moves[expense_sheet.id].line_ids[2].account_id)
```

Comment:
It's better to not rely on the order of lines (without sorting), also it would be better to explicitly check the lines account